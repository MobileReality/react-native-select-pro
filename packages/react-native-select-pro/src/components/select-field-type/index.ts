export { SelectFieldType } from './select-field-type';
