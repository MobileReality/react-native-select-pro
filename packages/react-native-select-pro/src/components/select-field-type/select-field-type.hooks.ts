import { useSelectContext } from '../../context';
import { selectedOptionResolver } from '../../helpers';

export const useSelectFieldType = () => {
    const { multiple, searchValue, selectedOption } = useSelectContext();

    const { selectedOptionLabel, selectedOptions } = selectedOptionResolver(selectedOption);

    const isSearchable = typeof searchValue === 'string';

    return {
        multiple,
        searchValue,
        selectedOptionLabel,
        selectedOptions,
        isSearchable,
    };
};
