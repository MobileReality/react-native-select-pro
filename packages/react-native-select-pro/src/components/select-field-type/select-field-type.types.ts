import type { OnPressRemove } from '../../types/shared';
import type { SelectControlStyles } from '../../types/styles';

export type SelectFieldTypeProps = {
    onPressRemove: OnPressRemove;
    selectStyles: SelectControlStyles | undefined;
};
