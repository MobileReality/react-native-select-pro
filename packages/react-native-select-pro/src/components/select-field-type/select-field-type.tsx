import React from 'react';
import type { ViewStyle } from 'react-native';
import { StyleSheet, View } from 'react-native';

import { MultiSelect } from '../multi-select';
import { SelectInput } from '../select-input';
import { SelectText } from '../select-text';

import { useSelectFieldType } from './select-field-type.hooks';

export const SelectFieldType = () => {
    const { multiple, selectedOptions, selectedOptionLabel, isSearchable } = useSelectFieldType();

    const renderSelectFieldType = () => {
        if (multiple) {
            return <MultiSelect selectedOptions={selectedOptions} />;
        }

        if (isSearchable) {
            return <SelectInput />;
        }

        return <SelectText selectedOptionLabel={selectedOptionLabel} />;
    };

    return <View style={styles.container}>{renderSelectFieldType()}</View>;
};

type Styles = {
    container: ViewStyle;
};

const styles = StyleSheet.create<Styles>({
    container: {
        flex: 1,
        height: '100%',
        justifyContent: 'center',
    },
});
