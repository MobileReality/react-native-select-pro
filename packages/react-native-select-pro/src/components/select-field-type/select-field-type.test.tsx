import React from 'react';

import { SelectFieldType } from './select-field-type';

describe('SelectFieldType', () => {
    it('should generate SelectFieldType snapshot', () => {
        const selectFieldType = (
            <SelectFieldType
                isOpened={false}
                selectedOption={null}
                dispatch={() => {}}
                disabled={false}
                multiSelection={false}
                placeholderText={undefined}
                placeholderTextColor={undefined}
                searchable={false}
                searchPattern={undefined}
                textInputProps={{}}
                searchValue=""
                setPosition={() => {}}
                textStyle={undefined}
                containerStyle={undefined}
                multiSelectionOptionStyle={undefined}
                onPressSelectControl={() => {}}
                onPressRemove={() => {}}
            />
        );
        expect(selectFieldType).toMatchSnapshot();
    });
});
