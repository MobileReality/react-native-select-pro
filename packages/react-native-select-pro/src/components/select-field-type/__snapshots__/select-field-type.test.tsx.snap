// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldType should generate SelectFieldType snapshot 1`] = `
<SelectFieldType
  disabled={false}
  dispatch={[Function]}
  isOpened={false}
  multiSelection={false}
  onPressRemove={[Function]}
  onPressSelectControl={[Function]}
  searchValue=""
  selectedOption={null}
  setPosition={[Function]}
  textInputProps={{}}
/>
`;
